pkg/profiling/profiling_test.go

```diff
@@ -0,0 +1,79 @@
+package profiling
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"sort"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+)
+
+func Test_fifoDir(t *testing.T) {
+	dir := t.TempDir()
+	maxFileCount := 3
+	fs := fifoDir{dirPath: dir, maxFileCount: maxFileCount}
+
+	// This should be bigger than maxFileCount to test FIFO deletion is done properly
+	numFilesToCreate := 10
+	for i := 0; i < numFilesToCreate; i++ {
+		fs.Create(fmt.Sprintf("%d.test.dump", i))
+	}
+
+	actualFiles, err := ioutil.ReadDir(dir)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if len(actualFiles) != maxFileCount {
+		t.Fatalf("expected count of files in test directory: %d, got %d", maxFileCount, len(actualFiles))
+	}
+
+	sort.Slice(actualFiles, func(i, j int) bool {
+		return actualFiles[i].ModTime().Before(actualFiles[j].ModTime())
+	})
+
+	expectedIndex := numFilesToCreate - maxFileCount
+	for _, f := range actualFiles {
+		parts := strings.Split(f.Name(), ".")
+		if len(parts) < 1 {
+			t.Fatal("expected filename to have index prefix seperated by '.'")
+		}
+
+		if fmt.Sprintf("%d", expectedIndex) != parts[0] {
+			t.Fatalf("expected index of file: %s to be: %d", f.Name(), expectedIndex)
+		}
```

Comment:
This seems quite complex and does not actually check that the right files are deleted. Perhaps just pre-generate the file names at the beginning of the test, then create them, and then compare  sorted list to a sub-slice of the generated names?

Reply:
Changed it to contains logic, which is in fact much simpler when using testify assert.